src/commands/client/utils.test-d.ts

```diff
@@ -0,0 +1,19 @@
+export const expectType = <T>(_: T) => {
+  // Do nothing, the TypeScript compiler handles this for us
+};
+
+export type TypeEqual<Target, Value> = (<T>() => T extends Target ? 1 : 2) extends <
+  T,
+>() => T extends Value ? 1 : 2
+  ? true
+  : false;
+
+export type NonArrayType<T> = T extends Array<infer U> ? U : T;
+
+export type NonNullable<T> = T extends null | undefined ? never : T;
```

Comment:
It seems that we don't use this one anywhere, and there's already an utility type for this. Could we maybe delete this?